frontend/packages/ux-editor/src/components/config/editModal/EditOptions/EditCodeList/OptionListEditor.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { createRef, useState } from 'react';
+import type { IGenericEditComponent } from '@altinn/ux-editor/components/config/componentConfig';
+import type { SelectionComponentType } from '@altinn/ux-editor/types/FormComponent';
+import type { Option } from 'app-shared/types/Option';
+import { useTranslation } from 'react-i18next';
+import {
+  StudioCodeListEditor,
+  StudioModal,
+  StudioSpinner,
+  StudioErrorMessage,
+} from '@studio/components';
+import { TableIcon } from '@studio/icons';
+import { useStudioEnvironmentParams } from 'app-shared/hooks/useStudioEnvironmentParams';
+import { useUpdateOptionListMutation } from '../../../../../hooks/mutations/useUpdateOptionListMutation';
+import { useOptionListsQuery } from '../../../../../hooks/queries/useOptionListsQuery';
+import { useOptionListEditorTexts } from './hooks/useOptionListEditorTexts';
+import { usePreviewContext } from 'app-development/contexts/PreviewContext';
+import classes from './OptionListEditor.module.css';
+
+type OptionListEditorProps = Pick<IGenericEditComponent<SelectionComponentType>, 'component'>;
+
+export function OptionListEditor({ component }: OptionListEditorProps): React.ReactNode {
+  const { t } = useTranslation();
+  const { org, app } = useStudioEnvironmentParams();
+  const { data: optionsListMap, status } = useOptionListsQuery(org, app);
+
+  switch (status) {
+    case 'pending':
+      return (
+        <StudioSpinner spinnerTitle={t('ux_editor.modal_properties_code_list_spinner_title')} />
+      );
+    case 'error':
+      return (
+        <StudioErrorMessage>{t('ux_editor.modal_properties_error_message')}</StudioErrorMessage>
+      );
+    case 'success': {
+      return (
+        <OptionListEditorModal
+          optionList={optionsListMap.get(component.optionsId)}
+          component={component}
+        />
+      );
+    }
+  }
+}
+
+type OptionListEditorModalProps = {
+  optionList: Option[];
+} & Pick<IGenericEditComponent<SelectionComponentType>, 'component'>;
+
+function OptionListEditorModal({
+  optionList,
+  component,
+}: OptionListEditorModalProps): React.ReactNode {
+  const { t } = useTranslation();
+  const { org, app } = useStudioEnvironmentParams();
+  const { doReloadPreview } = usePreviewContext();
+  const { mutate: uploadOptionList } = useUpdateOptionListMutation(org, app);
+  const [currentOptionList, setCurrentOptionList] = useState<Option[]>(optionList);
```

Comment:
And actually we do not need this state at all 🙈 
Just pass the incoming optionList to the `StudioCodeListEditor` and change handleOptionChange to this:
```
const handleOptionsChange = (options: Option[]) => {
    uploadOptionList({ optionListId: component.optionsId, optionsList: options });
  };
```

Reply:
We can introcude a debounce, and thereby not send a request to our backend for every keypress. 
Without the useState and sending the update during onClose, we get this error message from our backend:
```
fail: Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware[1]
      An unhandled exception has occurred while executing the request.
      System.IO.IOException: The process cannot access the file 'C:\Users\KONSIM\AppData\Local\altinn\repos\localgiteaadmin\localgiteaadmin\test\App\options\test text.json' because it is being used by another process.
         at Microsoft.Win32.SafeHandles.SafeFileHandle.CreateFile(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options)
         at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, Nullable`1 unixCreateMode)
         at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, Nullable`1 unixCreateMode)
         at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, Boolean useAsync)
         at Altinn.Studio.Designer.Infrastructure.GitRepository.GitRepository.WriteTextAsync(String absoluteFilePath, String text, CancellationToken cancellationToken) in C:\Users\KONSIM\Documents\Studio Repos\altinn-studio\backend\src\Designer\Infrastructure\GitRepository\GitRepository.cs:line 399
         at Altinn.Studio.Designer.Infrastructure.GitRepository.GitRepository.WriteTextByRelativePathAsync(String relativeFilePath, String text, Boolean createDirectory, CancellationToken cancellationToken) in C:\Users\KONSIM\Documents\Studio Repos\altinn-studio\backend\src\Designer\Infrastructure\GitRepository\GitRepository.cs:line 182
         at Altinn.Studio.Designer.Infrastructure.GitRepository.AltinnAppGitRepository.CreateOrOverwriteOptionsList(String optionsListId, List`1 payload, CancellationToken cancellationToken) in C:\Users\KONSIM\Documents\Studio Repos\altinn-studio\backend\src\Designer\Infrastructure\GitRepository\AltinnAppGitRepository.cs:line 778
         at Altinn.Studio.Designer.Services.Implementation.OptionsService.CreateOrOverwriteOptionsList(String org, String repo, String developer, String optionsListId, List`1 payload, CancellationToken cancellationToken) in C:\Users\KONSIM\Documents\Studio Repos\altinn-studio\backend\src\Designer\Services\Implementation\OptionsService.cs:line 62
         at Altinn.Studio.Designer.Controllers.OptionsController.CreateOrOverwriteOptionsList(String org, String repo, String optionsListId, List`1 payload, CancellationToken cancellationToken) in C:\Users\KONSIM\Documents\Studio Repos\altinn-studio\backend\src\Designer\Controllers\OptionsController.cs:line 127
         at lambda_method640(Closure, Object)
         at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(ActionContext actionContext, IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
         at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
         at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
         at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
         at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
         at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
      --- End of stack trace from previous location ---
         at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextExceptionFilterAsync>g__Awaited|26_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
         at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ExceptionContextSealed context)
         at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
         at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeNextResourceFilter()
      --- End of stack trace from previous location ---
         at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
         at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
         at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeFilterPipelineAsync()
      --- End of stack trace from previous location ---
         at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
         at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
         at Microsoft.AspNetCore.Localization.RequestLocalizationMiddleware.Invoke(HttpContext context)
         at Microsoft.AspNetCore.ResponseCompression.ResponseCompressionMiddleware.InvokeCore(HttpContext context)
         at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
         at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
         at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
         at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
         at Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddlewareImpl.<Invoke>g__Awaited|10_0(ExceptionHandlerMiddlewareImpl middleware, HttpContext context, Task task)
```

Which means we are sending too many requests for our backend to keep up because of file access management.

Reply:
Ah, did not see that. If we can use the useDebounce hook without too much hassle that would be a solution! But if that involves too much adaption I guess we can keep the state!

Reply:
I'll have a look tomorrow as i've not setup a debounce in our solution before.

Reply:
It would be preferable to use debounce as we do elsewhere in the solution, but if it introduces too much complexity I guess it is fine sticking with the useState in this case. We can consider creating an issue for it 🤷